bot/exts/events/trivianight/_questions.py

```diff
@@ -0,0 +1,256 @@
+
+    def set_questions(self, questions: list) -> None:
+        """Setting `self.questions` dynamically via a function to set it."""
+        self.questions = questions
+
+    def next_question(self, number: int = None) -> Union[Embed, None]:
+        """
+        Chooses a random unvisited question from the question bank.
+
+        If the number parameter is specified, it'll head to that specific question.
+        """
+        if all("visited" in question.keys() for question in self.questions) and number is None:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="All of the questions in the question bank have been used.",
+                color=Colours.soft_red
+            )
+
+        if number is None:
+            question_number = randrange(0, len(self.questions))
+            while "visited" in self.questions[question_number].keys():
+                question_number = randrange(0, len(self.questions))
+        else:
+            question_number = number - 1
+
+        self.questions[question_number]["visited"] = True
+        self.view.current_question = self.questions[question_number]
+
+    def list_questions(self) -> Union[Embed, str]:
+        """
+        Lists all questions from the question bank.
+
+        It will put the following into a message:
+            - Question number
+            - Question description
+            - If the question was already 'visited' (displayed)
+        """
+        if not self.questions:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No questions are currently loaded in!",
+                color=Colours.soft_red
+            )
+        spaces = len(
+            sorted(
+                self.questions, key=lambda question: len(question['description'].replace("\u200b", ""))
+            )[-1]["description"].replace("\u200b", "")
+        ) + 3
+        formatted_string = ""
+        for question in self.questions:
+            question_description = question["description"].replace("\u200b", "")
+            formatted_string += f"`Q{question['number']}: {question_description}" \
+                                f"{' ' * (spaces - len(question_description) + 2)}" \
+                                f"|` {':x:' if not question.get('visited') else ':white_check_mark:'}\n"
+
+        return formatted_string.strip()
+
+    def current_question(self) -> tuple[Embed, QuestionView]:
+        """Returns an embed entailing the current question as an embed with a view."""
+        return self.view.create_current_question(), self.view
+
+    def end_question(self) -> Embed:
+        """Terminates answering of the question and displays the correct answer."""
+        scores, answer_embed, time_limit, total_points = self.view.end_question()
+        for user, score in scores.items():
+            # Overhead with calculating scores leads to inflated times, subtracts 0.5 to give an accurate depiction
```

Comment:
It takes 0.5 seconds usually from the interaction handling on Discord's end.

Reply:
So if someone presses a button the instant Discord receives it they will still have taken 0.5 seconds according to us?

Isn't that just ping? I don't think we should be on relying on this - we aren't measuring exactly how fast people are to answer so that we can see who's the fastest, we just need to be able to give someone who clicked after 5 seconds more points than someone who clicked after 20.

Reply:
Good point, I'll remove the overhead subtracting